batch/batch/front_end/operators.py

```diff
@@ -0,0 +1,99 @@
+import abc
+from typing import List, Optional, Tuple
+
+
+class Operator(abc.ABC):
+    user_symbols: List[str]
```

Comment:
Are these meant to be class fields or instance fields? I suppose it probably doesn't matter because these classes are used as singletons, but I would expect that the `user_symbols` here be somehow defined as abstract and that subclasses are forced to define it. I honestly don't know how class fields work in this regard but an abstract property should work.